src/components/ShoppingCart/CartListItem.tsx

```diff
@@ -10,27 +10,48 @@ import Image from "next/image";
 interface CartListItemProps {
   item: ICartItem;
   onDelete: (id: string) => void;
+  checkedItems: boolean[];
+  setCheckedItems: Dispatch<SetStateAction<boolean[]>>;
+  index: number;
 }
 
-const CartListItem = ({ item, onDelete }: CartListItemProps) => {
+const CartListItem = ({
+  item,
+  onDelete,
+  checkedItems,
+  setCheckedItems,
+  index,
+}: CartListItemProps) => {
   const {
     design: { designName: productName, courtSize: courtDetail },
     quotation: quotation,
     image: image,
     isExpired,
   } = item;
 
+  const handleCheckBox = (e: { target: { checked: boolean } }) => {
+    const newCheckedItems = [...checkedItems];
+    newCheckedItems[index] = e.target.checked;
+    setCheckedItems(newCheckedItems);
+  };
```

Comment:
it can be replaced with
```
const handleCheckBox = () => {
  setCheckedItems(
    checkedItems.map((item, i) => {
      return index === i ? !item : item;
    })
  );
};
```